binary_port/src/binary_message.rs

```diff
@@ -74,25 +74,37 @@ impl codec::Decoder for BinaryMessageCodec {
     type Error = Error;
 
     fn decode(&mut self, src: &mut bytes::BytesMut) -> Result<Option<Self::Item>, Self::Error> {
-        let (length, have_full_frame) = if let [b1, b2, b3, b4, remainder @ ..] = &src[..] {
-            let length = LengthEncoding::from_le_bytes([*b1, *b2, *b3, *b4]) as usize;
-            (length, remainder.len() >= length)
-        } else {
-            // Not enough bytes to read the length.
-            return Ok(None);
-        };
-
-        if !have_full_frame {
-            // Not enough bytes to read the whole message.
-            return Ok(None);
-        };
+        let (length, remainder_length, have_full_frame) =
+            if let [b1, b2, b3, b4, remainder @ ..] = &src[..] {
+                let length = LengthEncoding::from_le_bytes([*b1, *b2, *b3, *b4]) as usize;
+                let remainder_length = remainder.len();
+                (length, remainder_length, remainder_length >= length)
+            } else {
+                // Not enough bytes to read the length.
+                return Ok(None);
+            };
 
         if length > self.max_message_size_bytes as usize {
             return Err(Error::RequestTooLarge {
                 allowed: self.max_message_size_bytes,
                 got: length as u32,
             });
         }
+
+        if remainder_length > self.max_message_size_bytes as usize {
```

Comment:
I don't think we need to do this as I mentioned on slack. In my view, the remainder could be either empty, or could contain partial frame, or multiple frames, or frame + another partial frame etc. I think in some extreme cases of very slow senders this may reject legitimate connections. This check above is sufficient (`length > self.max_message_size_bytes`).

Reply:
i agree...i would want proof that this is necessary.